packages/core-data/src/utils/on-sub-key.js

```diff
@@ -24,8 +30,25 @@ export const onSubKey = ( actionProperty ) => ( reducer ) => (
 		return state;
 	}
 
+	if ( action.type === 'REMOVE_ITEMS' ) {
```

Comment:
`onSubKey` is a generic higher order reducer which should not care about any particular action. This logic should be moved up to the reducer that uses `onSubKey`.

Reply:
This is the logic that removes items from the queries branch of the state.

Reply:
Yeah I got [a bit confused](https://github.com/WordPress/gutenberg/pull/22428#discussion_r428771569) but all is clear now. Will fix.